functors/README.md

```diff
@@ -0,0 +1,89 @@
+# Functors
+
+This directory contains examples of _functors_ with various strategies for distributed evaluation.
+
+## Identity functor
+
+[Locally evaluated identity functor](identity-functor.amb). 
+
+Equivalent source code (JavaScript):
+
+```javascript
+let identity = (value) => () => ({ value })
+let map_identity = () => (id, fn) => identity(fn(id().value))
+let string_length = (str) => str.length
+let program = () => map_identity()(identity("hello"), string_length)
+program()
+```
+
+Ambients encoding:
+
+```
+string_length[
+  in_ call.open call.(
+    func[
+      str[in_ arg.open arg.in int.in length.open_]|
+      int[
+        length[in_ str.open str]|
+        in_ str
+      ]|
+      open_
+    ]|
+    open return.open_
+  )
+]|
+map_identity[
+  in_ call.open call.(
+    func[
+      id[in_ arg.open arg.open identity.in arg.open_]|
+      arg[in_ id.open id.in func.in arg.open_]|
+      func[in_ func.in_ arg.open func.in identity.open_.open func]|
```

Comment:
Just to understand this: why `open func` after `open_`? Iirc we discussed that `open_` is always the last action.

Reply:
Good catch, I'll change that. It's nice though that with these hierarchical evaluation contexts, both versions do work ie. the converge the same even though at this particular level of evaluation the ordering is different.

Reply:
Yup, realized that too. And we actually discussed it (and iirc define in the paper) that "the remaining caps/ambients will be inhereted by the opening ambient" or smth like that. So this is a great example of it :)